### PR TITLE
feat: implements <Select /> (closes #6)

### LINKED ISSUE #6: Create <Select /> component
https://pattern-library.dequelabs.com/components/selects

### PR DESCRIPTION
**Pull request by schne324**

### CODE REVIEW

File: test/src/components/Select/index.js [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+    t.test('handles "down" arrow key', t => {
+      const wrapper = withDefaultSelected();
+      wrapper.setState({ expanded: true });
+      t.equal(wrapper.state('activeIndex'), 1);
+      wrapper.find('[role="combobox"]').simulate('keydown', { which: 40 });
+      t.equal(wrapper.state('activeIndex'), 2);
+      t.end();
+    });
+
+    t.test('handles "up" arrow key', t => {
```
Comment (dylanb) [inline test/src/components/Select/index.js]: nit: test the boundary conditions - up when at the top and down when at the bottom?

Comment (schne324) [reply test/src/components/Select/index.js]: 👍 I will add the boundary tests

File: demo/index.js [hunk 2]
```diff
@@ -78,6 +79,9 @@ const app = (
         <SideBarItem index={6}>
           <Link tabIndex={-1} to='/components/options-menu'>{'Options Menu'}</Link>
         </SideBarItem>
+        <SideBarItem index={7}>
+          <Link tabIndex={-1} to='/components/select'>{'Select'}</Link>
```
Comment (stephenmathieson) [inline demo/index.js]: This would look "nicer" as:

```js
<Link tabIndex={-1} to='/components/select'>Select</Link>
```

File: demo/patterns/components/Select/index.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+import { Select, SelectOption, Button } from 'src/';
+import './index.css';
+
+const Demo = () => (
+  <div className='select-demo'>
+    <h1>Select</h1>
+    <h2>Demo</h2>
+    <div>
+      <Select
+        label={'Day'}
```
Comment (stephenmathieson) [inline demo/patterns/components/Select/index.js]: `label='Day'`

File: src/lib/components/Select/index.js [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import keyname from 'keyname';
+import uniqueString from 'unique-string';
+import { search, shouldSearch } from './utils';
+export const SelectOption = props => (<div {...props} />);
+
+export default class Select extends Component {
+  static propTypes = {
+    children: PropTypes.node.isRequired,
```
Comment (stephenmathieson) [inline src/lib/components/Select/index.js]: Rather than passing the options as children, I'd suggest mirroring [react-select](https://github.com/JedWatson/react-select)'s API and just passing an `options=[...]` prop.

This would remove the need to for the `SelectOption` component entirely, and would require users to write less JSX in order to render a `<Select/>`.

File: src/lib/components/Select/index.js [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+    className: '',
+    required: false,
+    onKeyDown: () => {},
+    onSelect: () => {}
+  }
+
+  constructor() {
+    super();
+    this.state = {};
+    this.options = [];
```
Comment (stephenmathieson) [inline src/lib/components/Select/index.js]: Attaching the `options` to the instance is a bit concerning. What happens if an option is removed from the children array?

For example:

```js
class Foo extends React.Component {
  state = { items: [ 1, 2, 3, 4, 5 ] }

  render () {
    return <Select>{this.state.items.map(t => <SelectOption>{t}</SelectOption>
  }

  componentDidMount () {
    this.setState({ items: [ 6, 7, 8, 9, 10 ] })
  }
}
```

File: src/lib/components/Select/utils.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+let timer;
+let keys = [];
+
+/**
+ * Determines if, based on input, we should "search" the options for the given character
+ * @param  {Number} key     keyCode
+ * @return {Boolean}
+ */
+export const shouldSearch = key => {
+  var isLetter = key >= 65 && key <= 90;
```
Comment (stephenmathieson) [inline src/lib/components/Select/utils.js]: `const`

**Comment by schne324** [comment]
Hold off on merging this. The current implementation only supports an initially selected option. I'm working on a cleaner approach which supports programmatically selecting options.

Also the demo code sample needs to be updated to show the `options` prop

File: demo/index.js [hunk 7]
```diff
@@ -78,6 +79,9 @@ const app = (
         <SideBarItem index={6}>
           <Link tabIndex={-1} to='/components/options-menu'>{'Options Menu'}</Link>
```
Comment (schne324) [inline demo/index.js]: don't worry @stephenmathieson, I'll submit a separate PR that replaces any remaining text `{'Like this'}` with text `'Like this'` 😆

File: test/src/components/Select/index.js (hunk 1 above)
Comment (schne324) [reply test/src/components/Select/index.js]: [done](https://github.com/dequelabs/cauldron-react/pull/19/files#diff-05b3dd972ced7a87b82013c93f336ad4R230)
